src/app/services/serviceConfig.js

```diff
@@ -3,43 +3,22 @@ const he = require("he");
 const { Utils } = require("sequelize");
 const UrlValidator = require("login.dfe.validation/src/urlValidator");
 const logger = require("../../infrastructure/logger/index");
-const {
-  AUTHENTICATION_FLOWS,
-  GRANT_TYPES,
-  ACTIONS,
-  TOKEN_ENDPOINT_AUTH_METHOD,
-  ERROR_MESSAGES,
-  REDIRECT_URLS_CHANGES,
-} = require("../../constants/serviceConfigConstants");
+const { AUTHENTICATION_FLOWS, GRANT_TYPES, ACTIONS, TOKEN_ENDPOINT_AUTH_METHOD, ERROR_MESSAGES, REDIRECT_URLS_CHANGES } = require("../../constants/serviceConfigConstants");
 const { getServiceById } = require("../../infrastructure/applications");
-const {
-  getUserServiceRoles,
-  determineAuthFlowByRespType,
-  processRedirectUris,
-  processConfigurationTypes,
-  isCorrectProtocol,
-  isCorrectLength,
-  isValidUrl,
-  checkClientId,
-  _unescape,
-} = require("./utils");
-
-const {
-  saveRedirectUrlsToStorage,
-  deleteFromLocalStorage,
-  retreiveRedirectUrlsFromStorage,
-} = require("../../infrastructure/utils/serviceConfigCache");
+const { getUserServiceRoles, determineAuthFlowByRespType, processRedirectUris, processConfigurationTypes, isCorrectProtocol, isCorrectLength, isValidUrl, checkClientId, _unescape } = require("./utils");
+
+const { saveRedirectUrlsToStorage, deleteFromLocalStorage, retreiveRedirectUrlsFromStorage } = require("../../infrastructure/utils/serviceConfigCache");
 
 const buildServiceModelFromObject = (service, sessionService = {}) => {
   let tokenEndpointAuthMethod = null;
   const { CLIENT_SECRET_POST, CLIENT_SECRET_BASIC } = TOKEN_ENDPOINT_AUTH_METHOD;
   const sessionValue = sessionService?.tokenEndpointAuthMethod?.newValue;
   const fallbackValue = service.relyingParty.token_endpoint_auth_method === CLIENT_SECRET_POST ? CLIENT_SECRET_POST : CLIENT_SECRET_BASIC;
 
-  const responseTypes = (sessionService?.responseTypes?.newValue || service.relyingParty.response_types) || [];
+  const responseTypes = sessionService?.responseTypes?.newValue || service.relyingParty.response_types || [];
   const authFlowType = determineAuthFlowByRespType(responseTypes);
 
-  tokenEndpointAuthMethod = (sessionValue && (authFlowType === AUTHENTICATION_FLOWS.HYBRID_FLOW || authFlowType === AUTHENTICATION_FLOWS.AUTHORISATION_CODE_FLOW)) ? sessionValue : fallbackValue;
+  tokenEndpointAuthMethod = sessionValue && (authFlowType === AUTHENTICATION_FLOWS.HYBRID_FLOW || authFlowType === AUTHENTICATION_FLOWS.AUTHORISATION_CODE_FLOW) ? sessionValue : fallbackValue;
```

Comment:
Small suggestion (so small that you can opt into it if you want!), but maybe you could make a `isAuthFlowHybridOrAuthorisationCode` (or something like that) function, so then this line turns into 

```
const isAuthFlowHybridOrAuthorisationCode = isAuthFlowHybridOrAuthorisationCode(authFlow);
tokenEndpointAuthMethod = sessionValue && isAuthFlowHybridOrAuthorisationCode ? sessionValue : fallbackValue;
```

I always try to make ternary operators as easy to read as possible and I think that helps?  But maybe it's too wordy?  Your call ^^

Reply:
Or maybe reuse the
```
const isAuthorisationCodeFlow = authFlowType === AUTHENTICATION_FLOWS.AUTHORISATION_CODE_FLOW;`
``` 
code from further down in this file.  Something like
```
tokenEndpointAuthMethod = sessionValue && (isAuthorisationCodeFlow || isHybridFlow) ? sessionValue : fallbackValue;
```
is easier to read than the original.

Reply:
I think this is a good point and certainly helps with readability. I've made this change.